framework/src/test/java/org/tron/core/config/args/ArgsTest.java

```diff
@@ -112,5 +123,29 @@ public void get() {
         ByteArray.toHexString(Args.getLocalWitnesses()
             .getWitnessAccountAddress(CommonParameter.getInstance().isECKeyCryptoEngine())));
   }
+
+  @Test
+  public void testIpFromLibP2p()
+      throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+    Args.setParam(new String[] {"-w"}, Constant.TEST_CONF);
+
+    Config config = Configuration.getByFileName(null, Constant.TEST_CONF);
+    Config config2 = config.withoutPath(Constant.NODE_DISCOVERY_BIND_IP);
+    Config config3 = config2.withoutPath(Constant.NODE_DISCOVERY_EXTERNAL_IP);
+
+    CommonParameter.getInstance().setNodeDiscoveryBindIp(null);
+    CommonParameter.getInstance().setNodeExternalIp(null);
+
+    Method method1 = Args.class.getDeclaredMethod("bindIp", Config.class);
+    method1.setAccessible(true);
+    method1.invoke(Args.class, config3);
+
+    Method method2 = Args.class.getDeclaredMethod("externalIp", Config.class);
+    method2.setAccessible(true);
+    method2.invoke(Args.class, config3);
+
+    Assert.assertNotEquals("127.0.0.1", CommonParameter.getInstance().getNodeDiscoveryBindIp());
+    Assert.assertNotEquals("46.168.1.1", CommonParameter.getInstance().getNodeExternalIp());
```

Comment:
`"46.168.1.1"`: What kind of node is this IP?

Reply:
46.168.1.1 is the configed node.discovery.external.ip in Constant.TEST_CONF.
127.0.0.1 s the configed node.discovery.bind.ip in Constant.TEST_CONF.

Reply:
Then it is recommended to get the variables from the Constant.TEST_CONF

Reply:
I have updated this testcase.